### PR TITLE
api v1 - oauth support

### PR DESCRIPTION
**Pull request by pitr-ch**
- to use set and enable oauth in settings
- correcting typo

### CODE REVIEW

File: app/controllers/api/base_controller.rb [hunk 1]
```diff
@@ -50,29 +52,72 @@ def process_response(condition, response = nil)
...
+
+      unless auth.authorize
+        deny_access
+        return false
+      end
+
+      return true
+    end
+
+    class Authorize
```
Comment (ohadlevy) [inline app/controllers/api/base_controller.rb]: Why nested class? can't this go into its own file?

File: lib/foreman/default_settings/loader.rb [hunk 2]
```diff
@@ -62,6 +62,11 @@ def load(reset=false)
               set('use_uuid_for_certificates', "Should Foreman use random UUID's for certificate signing instead of hostnames", false),
               set('update_environment_from_facts', "Should Foreman update a host's environment from its facts", false)
             ].compact.each { |s| create s.update(:category => "Puppet")}
+
+            [ set('oauth_active', "Should foreman use OAuth for authorization in API", false),
+              set('consumer_key', "OAuth consumer key", 'katello'),
+              set('consumer_secret', "OAuth consumer secret", 'shhhh')
+            ].compact.each { |s| create s.update(:category => "OAuth")}
```
Comment (ohadlevy) [inline lib/foreman/default_settings/loader.rb]: maybe we should not call it oauth, but rather auth in general (as we have a few auth related settings)

**Comment by pitr-ch** [comment]
done and done

**Comment by ohadlevy** [comment]
that was merged a few days ago - thanks!

one question, can you please document how to use in the wiki?

**Comment by pitr-ch** [comment]
on my todo

**Pull request closed without merging**